hazelcast/src/main/java/com/hazelcast/cache/ICache.java

```diff
@@ -944,4 +944,12 @@
      */
     CacheStatistics getLocalCacheStatistics();
 
+    /**
+     * Opens cache if available (not destroyed).
+     * If cache is already open, does nothing.
+     *
+     * @throws java.lang.IllegalStateException if cache is already destroyed
+     */
+    void open();
```

Comment:
@mdogan can you look at the updated version of this PR with `ICacheInternal` that extends from `ICache` but servies `ICache` to API users. WDYT about this one ?